net.longturn.freeciv21.yaml

```diff
@@ -0,0 +1,89 @@
+# SPDX-FileCopyrightText: James Robertson <jwrober@gmail.com>
+#
+# Build Steps:
+#   - (once) sudo apt install flatpak-builder
+#   - (once) sudo flatpak remote-add --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
+#   - flatpak-builder --install-deps-from=flathub --force-clean build/flatpak net.longturn.freeciv21.yaml
+#   - flatpak-builder --user --install --force-clean build/flatpak net.longturn.freeciv21.yaml
+#   - flatpak run net.longturn.freeciv21
+#   - flatpak remove net.longturn.freeciv21
+#
+##########
+
+app-id: net.longturn.freeciv21
+runtime: org.kde.Platform
+runtime-version: '5.15-22.08'
+sdk: org.kde.Sdk
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --share=ipc
+  - --share=network
+  - --socket=pulseaudio
+  - --filesystem=home
+
+command: freeciv21-client
+modules:
+  # Get Lua 5.3 as a dependency not included in the org.kde.Platform package.
+  - name: Lua
+    sources:
+      - type: archive
+        url: https://www.lua.org/ftp/lua-5.3.6.tar.gz
+        sha256: fc5fd69bb8736323f026672b1b7235da613d7177e72558893a0bdcd320466d60
+        dest: include/lua5.3
+    buildsystem: simple
+    build-commands:
+      - cd include/lua5.3 && make linux
+      - cp -rv * ${FLATPAK_DEST}
+    cleanup:
+      - '*'
+
+  # Get Libertunus font as a dependency.
+  - name: Libertinus
+    sources:
+      - type: archive
+        url: https://github.com/alerque/libertinus/releases/download/v7.040/Libertinus-7.040.zip
+        sha256: 2cce08507441d8ae7b835cfe51fb643ad5d9f6b44db4360c4e244f0e474a72f6
+        dest: fonts/Libertinus
+    buildsystem: simple
+    build-commands:
+      - mkdir -p ${FLATPAK_DEST}/share/freeciv21/fonts/Libertinus/static/OTF
+      - cp -rv fonts/Libertinus/static/OTF/*.otf ${FLATPAK_DEST}/share/freeciv21/fonts/Libertinus/static/OTF
+
+  # Now build and package freeciv21
+  - name: freeciv21
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DCMAKE_BUILD_TYPE=Release
+      - -DFREECIV_DOWNLOAD_FONTS=OFF
+      - -DLUA_INCLUDE_DIR=/app/include/lua5.3/src
+      - -DLUA_LIBRARY=/app/include/lua5.3/src/liblua.a
+    cleanup:
+      - /include
+    sources:
+      - type: git
+        tag: v3.0-patch.1
+        url: https://github.com/longturn/freeciv21
+
+  # Finalize
+  - name: Finalize
+    buildsystem: simple
+    build-commands:
+      - mkdir -p ${FLATPAK_DEST}/share/icons/hicolor/64x64/apps
+      - cp ${FLATPAK_DEST}/share/freeciv21/misc/civicon.png ${FLATPAK_DEST}/share/icons/hicolor/64x64/apps/${FLATPAK_ID}.png
+      - cp ${FLATPAK_DEST}/share/freeciv21/misc/mpicon.png ${FLATPAK_DEST}/share/icons/hicolor/64x64/apps/net.longturn.freeciv21.modpack-qt.png
+      - mv ${FLATPAK_DEST}/share/applications/net.longturn.freeciv21.client.desktop ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
+      - sed -i 's/\/app\/share\/freeciv21\/misc\/civicon/net.longturn.freeciv21/' ${FLATPAK_DEST}/share/applications/${FLATPAK_ID}.desktop
```

Comment:
please use `destkop-file-edit` to modify desktop file.

Reply:
same for the lines below.

Reply:
btw sharing the same icon for the different desktop entries is bad UX (it's not blocking though)

Reply:
That is a cool utility. Yes agree bad UX using same icon. Maybe someone nice will make me a nice icon someday.